co_sim_io/impl/info.hpp

```diff
@@ -132,6 +138,52 @@ class Info
         return mOptions.size();
     }
 
+    void Save(std::ostream& out) const
+    {
+        static std::unordered_map<std::string, std::string> s_registered_object_names {
+            {typeid(Internals::InfoData<int>).name(),         "InfoData_int"},
+            {typeid(Internals::InfoData<double>).name(),      "InfoData_double"},
+            {typeid(Internals::InfoData<bool>).name(),        "InfoData_bool"},
+            {typeid(Internals::InfoData<std::string>).name(), "InfoData_string"}
+        };
+
+        out << Size() << "\n";
+        for (const auto& r_pair: mOptions) {
+            auto it_obj = s_registered_object_names.find(typeid(*(r_pair.second)).name());
+            CO_SIM_IO_ERROR_IF(it_obj == s_registered_object_names.end()) << "No name registered" << std::endl;
+            out << r_pair.first << "\n";
+            out << it_obj->second << "\n";
+            r_pair.second->Save(out);
+            out << "\n";
+        }
+    }
+    void Load(std::istream& in)
+    {
+        static std::unordered_map<std::string, std::shared_ptr<Internals::InfoDataBase>> s_registered_object_prototypes {
+            {"InfoData_int"    , std::make_shared<Internals::InfoData<int>>(1)},
+            {"InfoData_double" , std::make_shared<Internals::InfoData<double>>(1)},
+            {"InfoData_bool"   , std::make_shared<Internals::InfoData<bool>>(1)},
+            {"InfoData_string" , std::make_shared<Internals::InfoData<std::string>>("")}
+        };
+
+        std::string key, registered_name;
+
+        int size;
+        in >> size;
+
+        for (int i=0; i<size; ++i) {
+            in.ignore(10, '\n'); // skipping to end of line
```

Comment:
What are you ignoring here? And why max to 10?

Reply:
Hm you spotted the good one :)
So I have one word per line
I read it with >>, but this doesn’t go to the next line, which I read with getline
Probably now makes sense to use >> there as well
I will try

Reply:
now it works without this